src/qofonoextcellinfo.cpp

```diff
@@ -143,6 +153,10 @@ void QOfonoExtCellInfo::Private::getCellsAsync()
     connect(new QDBusPendingCallWatcher(iProxy->GetCellsAsync(), iProxy),
         SIGNAL(finished(QDBusPendingCallWatcher*)),
         SLOT(onGetCellsFinished(QDBusPendingCallWatcher*)));
+    if (!iActive) {
+        iActive = true;
```

Comment:
This looks suspicious. What I gather if one creates OfonoExtCellInfo in qml with a valid modem path and active set false, this will eventually override the active status when coming from the setModemPath() route.